src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Versioning/FhirCollectionSettingsUpdater.cs

```diff
@@ -73,6 +75,12 @@ public async Task ExecuteAsync(Container container, CancellationToken cancellati
                 thisVersion.Version = CollectionSettingsVersion;
                 await container.UpsertItemAsync(thisVersion, new PartitionKey(thisVersion.PartitionKey), cancellationToken: cancellationToken);
             }
+            else if (thisVersion.Version < CollectionSettingsVersion)
+            {
+                await _updateSP.Execute(container.Scripts, cancellationToken);
```

Comment:
I assume that if this step fails, then the FHIR service will fail to start, and it will be attempted again on the next startup.

What happens in a multi-instance scenario?  Do both instances try and update the values?

Reply:
There is a lock that happens before the FhirCollectionSettingsUpdater is called, so there can only be one. If it does fail then the version doesn't get updated and it will try to run again.